File: docs-examples/example-java/src/test/java/io/micronaut/configuration/rabbitmq/AbstractRabbitMQTest.java
package io.micronaut.configuration.rabbitmq;

import io.micronaut.context.ApplicationContext;
import org.testcontainers.containers.GenericContainer;
import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;

import java.util.HashMap;
import java.util.Map;

public abstract class AbstractRabbitMQTest {

    protected static GenericContainer rabbitContainer = new GenericContainer("library/rabbitmq:3.7")
            .withExposedPorts(5672)
            .waitingFor(new LogMessageWaitStrategy().withRegEx("(?s).*Server startup complete.*"));

    static {
        rabbitContainer.start();
    }

    protected ApplicationContext startContext() {
        Map<String, Object> config = new HashMap<>();
        config.put("rabbitmq.port", rabbitContainer.getMappedPort(5672));
        config.put("spec.name", this.getClass().getSimpleName());
        return ApplicationContext.run(config, "test");
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/AbstractRabbitMQTest.kt
package io.micronaut.rabbitmq

import io.kotest.core.spec.style.BehaviorSpec
import io.micronaut.context.ApplicationContext
import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy

abstract class AbstractRabbitMQTest(body: BehaviorSpec.() -> Unit): BehaviorSpec(body) {

    companion object {
        val rabbitContainer = KGenericContainer("library/rabbitmq:3.11.9-management")
                .withExposedPorts(5672)
                .waitingFor(LogMessageWaitStrategy().withRegEx("(?s).*Server startup complete.*"))!!

        init {
            rabbitContainer.start()
        }

        fun startContext(specName: String): ApplicationContext =
            ApplicationContext.run(getDefaultConfig(specName), "test")

        fun startContext(configuration: Map<String, Any>): ApplicationContext =
            ApplicationContext.run(configuration, "test")

        fun getDefaultConfig(specName: String): MutableMap<String, Any> =
            mutableMapOf(
                    "rabbitmq.port" to rabbitContainer.getMappedPort(5672),
                    "spec.name" to specName)
    }
}
